packages/account/src/Components/additional-kyc-info-modal/additional-kyc-info-form.tsx

```diff
@@ -0,0 +1,240 @@
+        const { client, ui, notifications } = useStore();
+        const { account_settings, residence_list } = client;
+        const { fields, initialValues, validate } = getFormConfig({
+            account_settings,
+            residence_list,
+            required_fields: ['place_of_birth', 'tax_residence', 'tax_identification_number', 'account_opening_reason'],
+        });
+
+        const {
+            update,
+            mutation: { isLoading, isSuccess, error, isError },
+        } = useSettings();
+
+        const onSubmit = async (values: typeof initialValues) => {
+            const tax_residence = residence_list.find(item => item.text === values.tax_residence)?.value;
+            const place_of_birth = residence_list.find(item => item.text === values.place_of_birth)?.value;
+
+            await update({ ...values, tax_residence, place_of_birth });
+        };
+
+        React.useEffect(() => {
+            if (!isLoading && isSuccess) {
+                notifications.refreshNotifications();
+                ui.toggleAdditionalKycInfoModal();
+                ui.toggleInformationSubmittedModal();
+            }
+        }, [isLoading, isSuccess, notifications, ui]);
+
+        React.useEffect(() => {
+            if (isError) {
+                setError(error as string);
+            }
+        }, [error, isError, setError]);
+
+        const onItemSelection =
+            (
+                field: string,
+                setFieldValue: (
+                    field: string,
+                    value: string,
+                    shouldValidate?: boolean | undefined
+                ) => Promise<void | FormikErrors<Record<string, unknown>>>
+            ) =>
+            ({ value, text }: TListItem) => {
+                setFieldValue(field, value ? text : '', true);
+            };
+
+        return (
+            <Formik
+                validateOnMount
+                validateOnBlur
+                validateOnChange
+                initialValues={initialValues}
+                onSubmit={onSubmit}
+                validate={validate}
+            >
+                {({ isValid, dirty, setFieldValue }) => (
+                    <Form className='additional-kyc-info-modal__form-layout'>
+                        {isLoading ? (
+                            <Loading is_fullscreen={false} />
+                        ) : (
+                            <section className='additional-kyc-info-modal__form-layout--fields'>
+                                <FormTitle />
+                                <fieldset className='additional-kyc-info-modal__form-field'>
+                                    <Field name='place_of_birth'>
+                                        {({ field, meta: { touched, error } }: FieldProps<typeof initialValues>) => (
+                                            <React.Fragment>
+                                                <DesktopWrapper>
+                                                    <Autocomplete
```

Comment:
Maybe we can create separate DropDownField and move the block of code to this separate component? 
I mean the block of Autocomplete and SekectNative with Desktop and Mobile wrappers

Reply:
We'll handle such refactoring in a card dedicated to personal-details-form after discussion if it's fine?